addons/calendar/models/calendar.py

```diff
@@ -0,0 +1,1047 @@
+            attendees = event.attendee_ids.mapped("partner_id")
+            new_attendees_ids = []
+            new_att_partner_ids = []
+            for partner in event.partner_ids:
+                if partner in attendees:
+                    continue
+                values = {
+                    'partner_id': partner.id,
+                    'event_id': event.id,
+                    'access_token': self.new_invitation_token(),
+                    'email': partner.email,
+                }
+
+                if partner.id == self.env.user.partner_id.id:
+                    values['state'] = 'accepted'
+
+                attendee = self.env['calendar.attendee'].create(values)
+                new_attendees_ids.append(attendee.id)
+                new_att_partner_ids.append(partner.id)
+
+                if not self.env.user.email or self.env.user.email != partner.email:
+                    mail_from = self.env.user.email or tools.config.get('email_from', False)
+                    if not self.env.context.get('no_email'):
+                        attendee._send_mail_to_attendees(email_from=mail_from)
+
+            if self.env.user.partner_id.id not in new_att_partner_ids:
+                values = {
+                    'partner_id': self.env.user.partner_id.id,
+                    'event_id': event.id,
+                    'access_token': self.new_invitation_token(),
+                    'email': self.env.user.partner_id.email,
+                }
+                current_user_att_id = self.env['calendar.attendee'].create(values)
+                new_attendees_ids.append(current_user_att_id.id)
+                new_att_partner_ids.append(self.env.user.partner_id.id)
+            if new_attendees_ids:
+                event.write({'attendee_ids': [(4, att) for att in new_attendees_ids]})
+            if new_att_partner_ids:
+                event.message_subscribe(new_att_partner_ids)
+
+            # We remove old attendees who are not in partner_ids now.
+            all_partner_ids = [part.id for part in event.partner_ids]
+            all_part_attendee_ids = [att.partner_id.id for att in event.attendee_ids]
+            all_attendee_ids = [att.id for att in event.attendee_ids]
+            partner_ids_to_remove = map(lambda x: x, set(all_part_attendee_ids + new_att_partner_ids) - set(all_partner_ids))
+            attendee_ids_to_remove_ids = []
+
+            if partner_ids_to_remove:
+                attendee_ids_to_remove = self.env["calendar.attendee"].search([('partner_id.id', 'in', partner_ids_to_remove), ('event_id.id', '=', event.id)])
+                attendee_ids_to_remove_ids.append(attendee_ids_to_remove.id)
+                if attendee_ids_to_remove:
+                    attendee_ids_to_remove.unlink()
+
+            res[event.id] = {
+                'new_attendee_ids': new_attendees_ids,
+                'old_attendee_ids': all_attendee_ids,
+                'removed_attendee_ids': attendee_ids_to_remove_ids
```

Comment:
Why not returning browse recordset instead of list, so that later wherever it is used we don't need to convert it to set, we can directly do intersection etc.
you may do like:
NewAttendees = self.env['calendar.attendee']
and later you can add record using
NewAttendees += attendee